array.c

```diff
@@ -2600,20 +2636,45 @@ ary_enum_length(VALUE ary, VALUE args, VALUE eobj)
 
 /*
  *  call-seq:
- *     ary.each {|item| block}    -> ary
- *     ary.each                   -> Enumerator
+ *    array.each {|element| ... } -> self
+ *    array.each -> Enumerator
  *
- *  Calls the given block once for each element in +self+, passing that element
- *  as a parameter.  Returns the array itself.
+ *  Iterates over array elements.
  *
- *  If no block is given, an Enumerator is returned.
+ *  ---
  *
- *     a = [ "a", "b", "c" ]
- *     a.each {|x| print x, " -- " }
+ *  When a block given, passes each successive array element to the block;
+ *  returns +self+:
```

Comment:
Should we also mention the behavior with `break` that causes #each to not return `self`?

```
$ ruby -e 'a = %w[a b].each { |element| break 5 }; p a'
5
```

Maybe you have a plan for that somewhere else? I'm not sure it fits here, especially because it is an unusual "gotcha" of ruby.

Reply:
Agree, not here.  Maybe someplace in doc/syntax?